test/Test/Core/Issue.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Test.Core.Issue where
+
+import IW.App (AppEnv)
+import IW.Core.Id (Id (..))
+import IW.Core.Issue (Issue (..))
+import IW.Core.Repo (RepoName (..), RepoOwner (..))
+import IW.Core.SqlArray (SqlArray (..))
+import IW.Effects.Log (runAppLogIO)
+import IW.Db.Functions (asSingleRow, query)
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+
+issueInsertAndQuery :: AppEnv -> Property
+issueInsertAndQuery env = property $ do
+  generatedIssue <- forAll genIssue
+  parsedIssue <- liftIO 
+              $ runAppLogIO env  
+              $ asSingleRow
+              $ query [sql| SELECT ?, ?, ?, ?, ?, ?, ?, ?|] generatedIssue
```

Comment:
Yeah, we almost there. Just one little tweak left. The problem now is because `fromRow` for `Issue` expects `Id` but `toRow` doesn't put `Id`. But we also don't want `toRow` to put `Id`! So the solution is to put the `Id` inside `SELECT`. Fortunately, this can be done relatively easy using data type composition from `postgresql-simple`:

```haskell
query [sql| SELECT ?, ?, ?, ?, ?, ?, ?, ?|] (Only (issueId generatedIssue) :. generatedIssue)
```

Yeah, at this point I think that our property-based roundtrip testing with `posgresql-simple` deserves a separate blog post...

Reply:
I agree! It's an interesting problem.

Reply:
I'm going to try to implement this.